src/api/MainSolver.h

```diff
@@ -68,136 +51,162 @@ const sstat s_Error = toSstat( 2);
 
 class MainSolver
 {
-  protected:
+protected: /** Helper classes to deal with assertion stack, preprocessing and substitutions **/
+    struct FrameId
+    {
+        uint32_t id;
+        bool operator== (const FrameId other) const { return id == other.id; }
+        bool operator!= (const FrameId other) const { return id != other.id; }
+    };
+
+    struct PushFrame {
```

Comment:
```suggestion
    struct PushFrame 
    {
```

Reply:
You are right that we are inconsistent in the style.
I think the style that we use elsewhere is to have opening brace at the same line as class name.
We will catch these style problems when we add it to clang-format files.

Reply:
Ok!